apisix/plugins/proxy-rewrite.lua

```diff
@@ -185,6 +185,7 @@ function _M.rewrite(conf, ctx)
         upstream_uri = core.utils.uri_safe_encode(upstream_uri)
     end
 
+    ctx.var.upstream_uri = nil
```

Comment:
If this line is added, it seems that `upstream_uri` will be set to nil whenever the `proxy-rewrite` plugin is used.

It doesn't look like a good fix. We should consider multiple scenarios.

Reply:
maybe ctx.var.upstream_uri = nil is fine ? because there will be unconditionally assigned to ctx.var.upstream_uri again。see lines 189 to 197

Reply:
it DOES NOT a good fix, but works

actually there are two problems to fix:
1. `proxy-rewrite` should not set `ctx.var.upstream_uri` when the `uri` or `regex_uri` isn't set
2. `ngx.var.xxx` can be set once at the time when `ctx.var.xxx` is first assign, after the metamthod `__newIndex` run , other assignment won't pass to the `ngx.var.xxx`

Reply:
The fix LGTM ( I can't give a better fix yet). Would be better to add a comment that we purge the cache by assigning a nil value.

Reply:
> @panhow We need some test cast cases to verify the changes are effective.

@panhow 
You can add the test in https://github.com/apache/apisix/blob/master/t/plugin/proxy-rewrite3.t.
You can refer to https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md to write and run the test locally.